### PR TITLE
fix(fa): updated the way bad refresh tokens are handled, added tests

### PR DESCRIPTION
**Pull request by akustan**

### CODE REVIEW

**Review by brennan-karrer — APPROVED** [review]

File: lib/fusion_auth/plugs/authorize_jwt.ex [hunk 1]
```diff
@@ -62,21 +63,34 @@ defmodule FusionAuth.Plugs.AuthorizeJWT do
...
+             {_, refresh} <- Utils.fetch_refresh(conn),
+             {:ok, %{"token" => new_token}, _} <-
+               FusionAuth.JWT.refresh_jwt(client, refresh, token) do
+          Plug.Conn.put_resp_header(
+            conn,
+            Application.get_env(:fusion_auth, :token_header_key),
+            new_token
+          )
+        else
+          {:error, body, env} ->
```
Comment (namster75) [inline lib/fusion_auth/plugs/authorize_jwt.ex:76]: There are three function calls in the `with` statements, are we sure we handle all the error cases for each of these function calls in the `else` block?

Comment (akustan) [reply lib/fusion_auth/plugs/authorize_jwt.ex:76]: yeah, I wrote tests for each case

Comment (namster75) [reply lib/fusion_auth/plugs/authorize_jwt.ex:76]: should it log the error for those error cases?

Comment (brennan-karrer) [reply lib/fusion_auth/plugs/authorize_jwt.ex:76]: Actually maybe we shouldn't be logging here from this package. Should we be logging from the consuming application for this package?

Comment (Maurice-Roy) [reply lib/fusion_auth/plugs/authorize_jwt.ex:76]: id generally opt for logging in the consuming package

Comment (brennan-karrer) [reply lib/fusion_auth/plugs/authorize_jwt.ex:76]: Yeah thinking that might be the better approach here

Comment (akustan) [reply lib/fusion_auth/plugs/authorize_jwt.ex:76]: hm okay, so how do we want to bubble up this info?

Comment (akustan) [reply lib/fusion_auth/plugs/authorize_jwt.ex:76]: it also looks like we already have fusionauth logging errors anytime a result status is >= 300

**Review by namster75 — APPROVED** [review]

**Pull request merged**